src/utils/Once.js

```diff
@@ -0,0 +1,17 @@
+class Once {
+  constructor(f) {
+    this.oncePromise = f().then(() => {
```

Comment:
So not lazy loading here I guess?

Reply:
Nah, I was just trying to make this as much like pthread_once as possible.  There's not really situations where I create the Once and don't immediately call "await once.do()" so I didn't see a reason to defer the Promise creation.

Reply:
Having said that if we think we want to use this for lazy initialization (create the Once in a constructor but defer actually running the code until later) I can make that change.  I'm unopinionated about it.